packages/image_picker/image_picker_platform_interface/lib/src/method_channel/method_channel_image_picker.dart

```diff
@@ -180,4 +180,92 @@ class MethodChannelImagePicker extends ImagePickerPlatform {
       type: retrieveType,
     );
   }
+
+  @override
+  Future<XFile?> getImage({
+    required ImageSource source,
+    double? maxWidth,
+    double? maxHeight,
+    int? imageQuality,
+    CameraDevice preferredCameraDevice = CameraDevice.rear,
+  }) async {
+    String? path = await _getImagePath(
+      source: source,
+      maxWidth: maxWidth,
+      maxHeight: maxHeight,
+      imageQuality: imageQuality,
+      preferredCameraDevice: preferredCameraDevice,
+    );
+    return path != null ? XFile(path) : null;
+  }
+
+  @override
+  Future<List<XFile>?> getMultiImage({
+    double? maxWidth,
+    double? maxHeight,
+    int? imageQuality,
+  }) async {
+    final List<dynamic>? paths = await _getMultiImagePath(
+      maxWidth: maxWidth,
+      maxHeight: maxHeight,
+      imageQuality: imageQuality,
+    );
+    if (paths == null) return null;
+
+    final List<XFile> files = [];
+    for (final path in paths) {
+      files.add(XFile(path));
+    }
+    return files;
+  }
+
+  @override
+  Future<XFile?> getVideo({
+    required ImageSource source,
+    CameraDevice preferredCameraDevice = CameraDevice.rear,
+    Duration? maxDuration,
+  }) async {
+    final String? path = await _getVideoPath(
+      source: source,
+      maxDuration: maxDuration,
+      preferredCameraDevice: preferredCameraDevice,
+    );
+    return path != null ? XFile(path) : null;
+  }
+
+  @override
+  Future<LostDataResponse> getLostData() async {
+    final Map<String, dynamic>? result =
+        await _channel.invokeMapMethod<String, dynamic>('retrieve');
+
+    if (result == null) {
+      return LostDataResponse.empty();
+    }
+
+    assert(result.containsKey('path') ^ result.containsKey('errorCode'));
```

Comment:
I need to ask internally if this bitwise-XOR works with booleans... or *how* it works (I'm guessing some implicit boolean -> int casting)... Anyway, this is non-documented behavior from what I could find in the language tour (and iffy)  :/

Reply:
I'm not actually sure of this myself, the contents of all of these methods are basically just 1:1 copies of their existing counterparts. I updated both instances of this to hopefully be a bit more conventional.

Reply:
Ah, yes, this is valid since Dart 2.1, according to the docs:

https://api.flutter.dev/flutter/dart-core/bool/operator_bitwise_exclusive_or.html #TIL